examples/cfd/07_cavity_flow.ipynb

```diff
@@ -0,0 +1,566 @@
+{
```

Comment:
<p>Can fix the error with an epsilon:</p><pre class="ql-syntax" spellcheck="false">l1norm = (np.sum(np.abs(_p.data[:]) - np.abs(_pn.data[:])) /
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; (np.sum(np.abs(_p.data[:])) + 1e-6))

</pre><p>You also can use l1-norm in numpy via <code>np.norm(_p.data[:], 1)</code></p><p> </p><p><code>oppres(p=_p, pn=_pn,time=3)</code>  any reason for <code>time_3</code>  or just empirical?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/devitocodes/devito/pull/1368/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
I subsequently had 
` if step not 0: `

I might switch to the np.norm.

time_3 is a debug leftover. The solutions wouldn't converge and I thought taking more smaller steps would help. I didn't.